integration-tests/deployment/keystone/deploy.go

```diff
@@ -724,25 +763,47 @@ func registerDons(lggr logger.Logger, req registerDonsRequest) (*registerDonsRes
 			return nil, fmt.Errorf("failed to confirm AddDON transaction %s for don %s: %w", tx.Hash().String(), don, err)
 		}
 		lggr.Debugw("registered DON", "don", don, "p2p sorted hash", p2pSortedHash, "cgs", cfgs, "wfSupported", wfSupported, "f", f)
+		registeredDons++
+	}
+	lggr.Debugf("Registered all DONS %d, waiting for registry to update", registeredDons)
+
+	// occasionally the registry does not return the expected number of DONS, so we retry a few times
```

Comment:
does it happen only right after update? or randomly..?

Reply:
only right after